test/lib/Symlink_test.js

```diff
@@ -77,4 +78,248 @@ describe('Symlink', function() {
+            beforeEach(function() {
+                this.id = '/foo/123';
+                this.data = {
+                    id: 123,
+                    bar: 'test bar',
+                    derp: true
+                };
+                this.symlink = new Symlink(this.id, this.data);
+                this.connection = {
+                    get: sinon.stub()
+                        .withArgs('/foo/123')
+                        .returns(Promise.resolve({
+                            flerp: 'test flerp'
+                        }))
+                };
+            });
+
+            describe('all requested props prehydrated', function() {
+                beforeEach(function() {
+                    this.props = ['id', 'bar', 'derp'];
+                });
+
+                it('skips internal API call', function() {
+                    return this.symlink.resolve(this.connection, { props: this.props })
+                    .then(function(result) {
+                        this.connection.get.called.should.be.false;
+                    }.bind(this));
+                });
+
+                it('resolves with resource', function() {
+                    return this.symlink.resolve(this.connection, { props: this.props })
+                    .then(function(result) {
+                        result.should.have.property('id', 123);
+                        result.should.have.property('bar', 'test bar');
+                        result.should.have.property('derp', true);
+                    }.bind(this));
+                });
+            });
+
+            describe('some requested props prehydrated', function() {
+                beforeEach(function() {
+                    this.props = ['id', 'bar', 'derp', 'flerp'];
+                });
+
+                it('makes internal API call with missing props', function() {
+                    return this.symlink.resolve(this.connection, { props: this.props })
+                    .then(function(result) {
+                        this.connection.get.called.should.be.true;
+                        this.connection.get.lastCall.args[0].should.equal(this.symlink.$link);
+                        this.connection.get.lastCall.args[1].should.have.property('props');
+                        this.connection.get.lastCall.args[1].props.should.contain('flerp');
+                    }.bind(this));
+                });
+
+                it('makes internal API call without prehydrated props', function() {
+                    return this.symlink.resolve(this.connection, { props: this.props })
+                    .then(function(result) {
+                        this.connection.get.lastCall.args[1].props.should.not.contain('id');
+                        this.connection.get.lastCall.args[1].props.should.not.contain('bar');
+                        this.connection.get.lastCall.args[1].props.should.not.contain('derp');
+                    }.bind(this));
```

Comment:
it'd be good to add a unit test where we check the result of the symlink resolution that it actually returned flerp = 'test flerp'